### ISSUE TITLE
Seed Analyser GUI won't Open

### ISSUE THREAD

**Issue by DorianGrayJr** [AgriCraft/AgriCraft#1542]
When trying to open the seed analyser GUI it won't open no matter what. I looked up for a version of InfinityLib as I saw that fixes the issue but is not available on the version I'm playing.

Mods and versions used

* Minecraft: 1.20.1 forge 47.3.0
* AgriCraft: 1.20.1-4.0.2-beta
* InfinityLib: not available for this version

I'm currently on single player. Also the pam harvestcraft doesn`t have texture after you get them after planting them
![image](https://github.com/user-attachments/assets/deb73642-ff49-484b-a032-8075e605447d)

**Comment by Terrails** [comment]
Can confirm the same issue. The logs contain an exception at this line due to `data` being `null`.
https://github.com/AgriCraft/AgriCraft/blob/a87f6d439c63b94c28d40f66acbf6921c3a3fe74/common/src/main/java/com/agricraft/agricraft/common/registry/ModMenus.java#L15

**Comment by TheGameAce** [comment]
Adding to this. Came here specifically to report the same issues. Seed Analyzer also gets broken quite slowly in addition to the above. Also adding that the analyzer is completely bugged out and consumes seeds put into it, while duplicating the agricultural journal.

**Comment by SacrificialGoatTwin** [comment]
Can confirm this too, journal is duplicated but the duplicate is a ghost item and does vanish sometimes. Same with seeds although seeds do actually dupe sometimes.

**Comment by alchemyyy** [comment]
Here's a disgusting little fix. seems to work in the 10 seconds of testing I did.

```
public static final PlatformRegistry.Entry<MenuType<SeedAnalyzerMenu>> SEED_ANALYZER_MENU = MENUS.register("seed_analyzer",
			() -> Platform.get().createMenuType((id, inv, data) -> new SeedAnalyzerMenu(id, inv, inv.player, GetLookingAtBlock(inv.player))));

public static BlockPos GetLookingAtBlock(Player player){
        HitResult block =  player.pick(20.0D, 0.0F, false);
		if(block.getType() == HitResult.Type.BLOCK)
		{
			return ((BlockHitResult)block).getBlockPos();
		}
		return null;
	}`

**Comment by Terrails** [comment]
The actual solution from looking it up is to replace the call to
https://github.com/AgriCraft/AgriCraft/blob/a87f6d439c63b94c28d40f66acbf6921c3a3fe74/forge/src/main/java/com/agricraft/agricraft/common/util/forge/ForgePlatform.java#L131
With a call to `NetworkHooks.openScreen`.

**Comment by Ketheroth** [comment]
Should be fixed with version 4.0.3
